### PR TITLE
EL18 RF USB switch

### PR DESCRIPTION
**Pull request by gagarinlg**
Summary of changes:
The USB connection can switched between the MCU and the internl RF module from a menu in the bootloader
Also it is now possible to add radio specific menu entries to the bootloader main menu

### CODE REVIEW

**Comment by gagarinlg** [comment]
A nice icon for the menu lines would be nice. Any suggestions?

**Comment by gagarinlg** [comment]
After doing some more testing, the board type detection does not work correctly, NV14s are also detectd as EL18 by the bootloader, but the EL18 part can be tested

**Comment by gagarinlg** [comment]
so somehow the changes are messing with the radio type detection in the firmware. The bootloader does detect the radio type, the firmware not. As soon as the changed bootloader is used, the firmware always detects NV14.
The access to the RF module and upgrading RXs can be tested

**Comment by pfeerick** [comment]
Something is still a bit fishy with that last commit... when I tried it the menu entry was missing. I then tried the prior commit (which as a merge commit comes down as 204ad95c4e8cf82e0edde6b356f647a0cf9d8764) and the entry shows there. 

With Flysky Assistant the module is detected, and I can 'start' doing a forced update of the receiver (which goes from slow triple flash to a faster triple flash), but it fails saying no response (with two different receivers which were previously updated using PL18EV). 

image

image

**Comment by gagarinlg** [comment]
On my radio the bootloader consistently detects the radio type correctly as EL18, but the firmware detects, also consistently, a NV14.
But I could not find a change, that I did that should trigger such a mess.

Maybe the internal module has to be enabled with tge enable pin, even though it is visible via USB.

**Comment by gagarinlg** [comment]
Current state, from within the bootloader:
two FTr10: upgrade and downgrade OK, connection to EL18 with newer firmware OK
FGr4S: upgrade and downgrade OK, connection to EL18 with newer firmware OK
FTr12B: downgrade OK, upgrade NOK, it seems like the newer software does not like my RX
Tr8B: upgrade and downgrade NOK, connection to EL18 OK

as a side note:
I enabled the RF module USB manually by editing the registers
The status disbalay on the module page does help when upgrading the receivers, as the "module Status" shows the current state of the upgrade process. 
The Assistant also displays additional reciver info, like the firmware version, when a reciever is bound and connected. which is quiet nice. There is no way to get the RX SW version with out that setup, currently.
Fiour possibilities here:
1. get the radio firmware to show the RX version
2. also allow the USB switch from the module page
3. add the module communication to the bootloader (I am not a fan of that option!) 
4. leave it as it is now

**Comment by richardclli** [comment]
> add the module communication to the bootloader (I am not a fan of that option!)

Well, agreed all functionalities specialized to manufacturers should not be hardcoded in ETX code base.  However, no harm to support via LUA scripts, just like ELRS configuration.

And the script, leave for others who are interested to work out ^.^

**Comment by raphaelcoeffic** [comment]
> get the radio firmware to show the RX version

that is coming, so we have it just like the other modules that support it.

File: radio/src/targets/nv14/board.h [hunk 1]
```diff
@@ -140,6 +144,13 @@ void EXTERNAL_MODULE_OFF();
 #define BLUETOOTH_MODULE_ON()           GPIO_ResetBits(BLUETOOTH_ON_GPIO, BLUETOOTH_ON_GPIO_PIN)
 #define BLUETOOTH_MODULE_OFF()          GPIO_SetBits(BLUETOOTH_ON_GPIO, BLUETOOTH_ON_GPIO_PIN)
 
+#if defined(BOOT)
+
+#define USB_SW_TO_INTERNAL_MODULE() GPIO_SetBits(USB_SW_GPOIO, USB_SW_PIN);
+#define USB_SW_TO_MCU() GPIO_ResetBits(USB_SW_GPOIO, USB_SW_PIN);
```
Comment (raphaelcoeffic) [inline radio/src/targets/nv14/board.h:150]: I think these 2 could be solely in the `.cpp` code. No need to export anything here.

Comment (gagarinlg) [reply radio/src/targets/nv14/board.h]: done

**Comment by rotorman** [comment]
Tested with two specimens, both work as expected:
grafik

**Review by pfeerick — APPROVED** [review]
Both my NV14 and EL18 are working correct in both bootloader and main firmware - both UI and RF link. Feel free to merge if I go offline again as lost internet for part of the day.

**Pull request merged**